sklearn/tree/_criterion.pyx

```diff
@@ -994,24 +910,20 @@ cdef class MSE(RegressionCriterion):
         The absolute impurity improvement is only computed by the
         impurity_improvement method once the best split has been found.
         """
-        cdef SIZE_t n_outputs = self.n_outputs
-
-        cdef double weighted_n_left = self.weighted_n_left
-        cdef double weighted_n_right = self.weighted_n_right
 
         cdef double* sum_left = self.sum_left
         cdef double* sum_right = self.sum_right
 
-        cdef SIZE_t k = 0
-        cdef double proxy_impurity_left = 0.
-        cdef double proxy_impurity_right = 0.
+        cdef SIZE_t k
+        cdef double proxy_impurity_left = 0.0
+        cdef double proxy_impurity_right = 0.0
 
-        for k in range(n_outputs):
-            proxy_impurity_left += sum_left[k] * sum_left[k]
-            proxy_impurity_right += sum_right[k] * sum_right[k]
+        for k in range(self.n_outputs):
+            proxy_impurity_left += sum_left[k] ** 2.0
+            proxy_impurity_right += sum_right[k] ** 2.0
```

Comment:
Everywhere else it's array[k] * array[k] so it would be better to be consistent.